src/handlers/web3.ts

```diff
@@ -0,0 +1,683 @@
+      : p.estimateGas(txPayloadToEstimate));
+
+    const lastBlockGasLimit = addBuffer(gasLimit.toString(), 0.9);
+    const paddedGas = addBuffer(
+      estimatedGas.toString(),
+      paddingFactor.toString()
+    );
+    logger.sentry('⛽ GAS CALCULATIONS!', {
+      estimatedGas: estimatedGas.toString(),
+      gasLimit: gasLimit.toString(),
+      lastBlockGasLimit: lastBlockGasLimit,
+      paddedGas: paddedGas,
+    });
+
+    // If the safe estimation is above the last block gas limit, use it
+    if (greaterThan(estimatedGas.toString(), lastBlockGasLimit)) {
+      logger.sentry(
+        '⛽ returning orginal gas estimation',
+        estimatedGas.toString()
+      );
+      return estimatedGas.toString();
+    }
+    // If the estimation is below the last block gas limit, use the padded estimate
+    if (greaterThan(lastBlockGasLimit, paddedGas)) {
+      logger.sentry('⛽ returning padded gas estimation', paddedGas);
+      return paddedGas;
+    }
+    // otherwise default to the last block gas limit
+    logger.sentry('⛽ returning last block gas limit', lastBlockGasLimit);
+    return lastBlockGasLimit;
+  } catch (error) {
+    logger.error('Error calculating gas limit with padding', error);
+    return null;
+  }
+}
+
+/**
+ * @desc convert from ether to wei
+ * @param value The value in ether.
+ * @return The value in wei.
+ */
+export const toWei = (ether: string): string => {
+  const result = parseEther(ether);
+  return result.toString();
+};
+
+/**
+ * @desc get transaction info
+ * @param hash The transaction hash.
+ * @return The corresponding `TransactionResponse`, or `null` if one could not
+ * be found.
+ */
+export const getTransaction = async (
+  hash: string
+): Promise<TransactionResponse | null> =>
+  web3Provider ? web3Provider.getTransaction(hash) : null;
+
+/**
+ * @desc get address transaction count
+ * @param address The address to check.
+ * @return The transaction count, or `null` if it could not be found.
+ */
+export const getTransactionCount = async (
+  address: string
+): Promise<number | null> =>
+  web3Provider ? web3Provider.getTransactionCount(address, 'pending') : null;
+
+/**
+ * @desc Gets transaction details for a new transaction.
+ * @param transaction The new transaction. In particular, the `from`, `to`,
+ * `gasPrice`, `gasLimit`, `amount` fields from a `NewTransaction` are required,
+ * as well as an optional `data` field similar to a `TransactionRequest`.
+ * @return The transaction details.
+ */
+export const getTxDetails = async (
+  transaction: Partial<
+    Pick<
+      NewTransactionNonNullable,
+      'from' | 'to' | 'gasPrice' | 'gasLimit' | 'amount'
+    >
+  > &
+    Pick<TransactionRequest, 'data'>
+): Promise<TransactionDetailsReturned> => {
+  const { to } = transaction;
+  const data = transaction.data ? transaction.data : '0x';
+  const value = transaction.amount ? toHex(toWei(transaction.amount)) : '0x00';
```

Comment:
```suggestion
  const value = toHex(toWei(transaction?.amount)) ?? '0x00';
```

I will ignore similar comments for the rest of the code